public/components/application_analytics/components/application.tsx

```diff
@@ -217,17 +214,32 @@ export function Application(props: AppDetailProps) {
   }, [appId]);
 
   useEffect(() => {
-    chrome.setBreadcrumbs([
-      ...parentBreadcrumbs,
-      {
-        text: 'Applications',
-        href: '#/',
-      },
-      {
-        text: application.name,
-        href: `#/${appId}`,
-      },
-    ]);
+    const isNavGroupEnabled = coreRefs?.chrome?.navGroup.getNavGroupEnabled();
+    if (isNavGroupEnabled) {
+      chrome.setBreadcrumbs([
+        {
+          text: 'Applications',
+          href: '#/',
+        },
+        {
+          text: application.name,
+          href: `#/${appId}`,
+        },
+      ]);
+    } else {
+      chrome.setBreadcrumbs([
+        ...parentBreadcrumbs,
+        {
+          text: 'Applications',
+          href: '#/',
+        },
+        {
```

Comment:
Can we please create a function like generateBreadcrumbs() that will remove the logic from all of these places and move it to somewhere that can be easily changed in the future if need be?

Reply:
Good call out, thanks added in setNavBreadCrumbs()